### DISCUSSION TITLE
Multipart Parser for Wado call

### DISCUSSION THREAD

**Discussion by Homersbm** [dcm4che/dcm4che#933] [Q&A] [answered]
I have  a Multipart Response that looks like this:
It has the boundry (--b5e6f7ea-797f-4918-8edc-5c5742315e60) plus some another 4 lines and then the body.
How would you manipulate the Existing MultiParser or MultiPartInputStream to get only the body
The current code only considers removing the Boundry and consuming the remaining as body

--b5e6f7ea-797f-4918-8edc-5c5742315e60
Content-Type: application/dicom
Content-Location: https://store/dicom/wadors/6a9d7c58-174b-4a42-afd3-10a07dbefd68/studies/1.3.46.670589.11.86.5.0.6860.2012012714363257048/series/1.3.46.670589.11.86.5.0.2932.2013012715122148150/instances/1.3.46.670589.11.86.5.0.2932.2013012715145835151
Content-Length: 255884

BODY DICM   UL ´     OB       UI 1.2.840.10008.5.1.4.1.1.4   UI0 1.3.46.670589.11.86.5.0.2932.2013012715145835151  UI 1.2.840.10008.1.2.1   UI 1.3.46.670589.54.2.20.7   SH 20.7.0.0  CS
--b5e6f7ea-797f-4918-8edc-5c5742315e60
Content-Type: application/dicom
Content-Location: https://store/dicom/wadors/6a9d7c58-174b-4a42-afd3-10a07dbefd68/studies/1.3.46.670589.11.86.5.0.6860.2012012714363257048/series/1.3.46.670589.11.86.5.0.2932.2013012715122148150/instances/1.3.46.670589.11.86.5.0.2932.2013012715145842152
Content-Length: 255884

------
I have overriden MultiParser and MultiPartInputStream classes to write my own Parsing logic but it hasnt come off yet.

Existing parsing LOGIC from DCM4che.
new MultiParser(boundry).parse(new BufferedInputStream(stream), new MultiParser.Handler() {
                @Override
                public void bodyPart(int partNumber, MultiPartInputStream var2) throws IOException {
                    try {
                        String fileName = fileName(partNumber, uid);
                        write(var2, fileName);
                    } catch (Exception e) {
                       
                    }
                }
				
				
	private static void write(InputStream in, String fileName) throws IOException {
        Path path = outDir != null
                ? Files.createDirectories(Paths.get(outDir)).resolve(fileName)
                : Paths.get(fileName);
        try (OutputStream out = Files.newOutputStream(path)) {
            StreamUtils.copy(in, out);
        }
    }

**Comment by gunterze** [comment, marked as answer]
Read [WadoRS.java](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-wadors/src/main/java/org/dcm4che3/tool/wadors/WadoRS.java) or [StowRSServer.java](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-stowrsd/src/main/java/org/dcm4che3/tool/stowrsd/StowRSServer.java) to see how to use `org.dcm4che3.mime.MultipartParser` and `org.dcm4che3.mime.MultipartInputStream` to unpack MIME parts from multipart messages.
